src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -229,15 +229,21 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 
             var builder = new DefaultTagHelperContent();
 
-            if (mode == Mode.Fallback && string.IsNullOrEmpty(HrefInclude) || mode == Mode.AppendVersion)
+            if (!string.IsNullOrEmpty(Href) && AppendVersion == true)
```

Comment:
Ah, it seems the `BuildGlobbedLinkTags` replaces the value of `href`, therefore the functional impact wont exist. Only benefit of not moving the `attributes` then is so that it reduces the number of attribute copies that occur when we duplicate the `output.Attributes` list.